flask_user/forms.py

```diff
@@ -210,7 +221,7 @@ def validate(self):
             user, user_email = user_manager.find_user_by_email(self.email.data)
 
         # Handle successful authentication
-        if user and user.password and user_manager.verify_password(self.password.data, user):
+        if user and user_manager.verify_password(self.password.data, user):
```

Comment:
If `self.password` does not exist, then you have clearly changed the original form class and it would make sense that you need to also override the form's `validate` method to accomodate your changes.